File: 30 Days of Code/C/Day23_BST_LevelOrder(BFS).c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

typedef struct Node{
    struct Node* left;
    struct Node* right;
    int data;
}Node;
Node* newNode(int data){
    Node* node=(Node*)malloc(sizeof(Node));
    node->left=node->right=NULL;
    node->data=data;
    return node;
}

    int max(int a, int b)
    {
        return a>b ? a : b;
    }
    
    int height(Node* root){
        if(root == NULL)return 0;
        
        return 1+max(height(root->left), height(root->right));
        
    }
    
    void Print(Node* root, int h)
    {
        if(root == NULL)return ;
        if(h == 1)printf("%d ", root->data);
        else if(h > 1){
            Print(root->left, h-1);
            Print(root->right, h-1);
        }
    }

    void levelOrder(Node * root){
        int h = height(root);
        for(int i=1; i<=h; i++){
            Print(root, i);
        }
    
    }

Node* insert(Node* root,int data){
    if(root==NULL)
        return newNode(data);
    else{
        Node* cur;
        if(data<=root->data){
            cur=insert(root->left,data);
            root->left=cur;                
        }
        else{
            cur=insert(root->right,data);
            root->right=cur;
        }
        
    }
    return root;
}
int main(){
    Node* root=NULL;
    int T,data;
    scanf("%d",&T);
    while(T-->0){
        scanf("%d",&data);
        root=insert(root,data);
    }
    levelOrder(root);
    return 0;
    
}

File: 30 Days of Code/Java/Day23_BST_LevelOrder(BFS).java
import java.util.*;
import java.io.*;
class Node{
    Node left,right;
    int data;
    Node(int data){
        this.data=data;
        left=right=null;
    }
}
class Solution{

public static int max(int a, int b)
    {
        return a>b ? a : b;
    }
    
    public static int height(Node root){
        if(root == null)return 0;
        
        return 1+max(height(root.left), height(root.right));
        
    }
    
    public static void Print(Node root, int h)
    {
        if(root == null)return ;
        if(h == 1)System.out.printf("%d ", root.data);
        else if(h > 1){
            Print(root.left, h-1);
            Print(root.right, h-1);
        }
    }

    public static void levelOrder(Node root){
        int h = height(root);
        for(int i=1; i<=h; i++){
            Print(root, i);
        }
    }

public static Node insert(Node root,int data){
        if(root==null){
            return new Node(data);
        }
        else{
            Node cur;
            if(data<=root.data){
                cur=insert(root.left,data);
                root.left=cur;
            }
            else{
                cur=insert(root.right,data);
                root.right=cur;
            }
            return root;
        }
    }
    public static void main(String args[]){
            Scanner sc=new Scanner(System.in);
            int T=sc.nextInt();
            Node root=null;
            while(T-->0){
                int data=sc.nextInt();
                root=insert(root,data);
            }
            levelOrder(root);
        }	
}